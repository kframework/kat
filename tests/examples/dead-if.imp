
int x ;
x = 7 ;
if (x <= 7) {
  x = 1 ;
} else {
  x = -1 ;
}
