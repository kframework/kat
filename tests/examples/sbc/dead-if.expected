Result ==K <kat-imp>
  <s>
    #STUCK ( ) ~> #compile-result .Rules , < { int x , .Ids ; x = 7 ; if ( x <= 7 ) { x = 1 ; } else { x = -1 ; } | .Map } --> { if ( x <= 7 ) { x = 1 ; } else { x = -1 ; } | x |-> 7 } > , < { false ~> if ( x <= 7 ) { x = 1 ; } else { x = -1 ; } | x |-> V0 } --> { . | x |-> -1 } > , < { true ~> if ( x <= 7 ) { x = 1 ; } else { x = -1 ; } | x |-> V0 } --> { . | x |-> 1 } >
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <imp>
    <k>
      .
    </k>
    <mem>
      x |-> V1
    </mem>
  </imp>
</kat-imp>
